src/components/Dialogs/UndelegateDialog/UndelegateDialog.tsx

```diff
@@ -46,25 +47,40 @@ export function UndelegateDialog({
     delegate.address.toLowerCase() === accountAddress?.toLowerCase();
 
   const isDisabledInTenant = ui.toggle("delegates/delegate")?.enabled === false;
-
-  const { data: accountEnsName } = useEnsName({
-    chainId: 1,
-    address: accountAddress as `0x${string}`,
-  });
+  const isGasRelayEnabled = ui.toggle("sponsoredDelegate")?.enabled === true;
 
   const { data: delegateeEnsName } = useEnsName({
     chainId: 1,
     address: delegatee?.delegatee as `0x${string}`,
   });
 
-  const { isError, writeContract: write, data } = useWriteContract();
+  const {
+    call,
+    isFetching: isProcessingSponsoredDelegation,
+    isFetched: didProcessSponsoredDelegation,
```

Comment:
NIT and not blocking but might be nice to change these to `isProcessingSponsoredUnDelegation`